src/github.com/getlantern/flashlight/client/handler.go

```diff
@@ -142,12 +154,12 @@ func (client *Client) intercept(resp http.ResponseWriter, req *http.Request) {
 // responsible for responding to the initial CONNECT request with a 200 OK.
 func pipeData(clientConn net.Conn, connOut net.Conn, closeFunc func()) {
 	// Start piping from client to proxy
-	go func() {
+	ctx.Go(func() {
```

Comment:
This is the special magic that's required for context to jump goroutine boundaries.  Use ctx.Go instead of go.